src/deluge/model/clip/instrument_clip.cpp

```diff
@@ -2393,62 +2373,23 @@ void InstrumentClip::writeDataToFile(Serializer& writer, Song* song) {
-		writer.write("\"0x");
-		intToHex(arpSettings.lastLockedSpreadVelocityParameterValue, buffer);
-		writer.write(buffer);
-		for (int i = 0; i < SPREAD_LOCK_MAX_SAVED_VALUES; i++) {
-			intToHex(arpSettings.lockedSpreadVelocityValues[i], buffer, 2);
-			writer.write(buffer);
-		}
-		writer.write("\"");
-		// Spread gate
-		writer.insertCommaIfNeeded();
-		writer.write("\n");
-		writer.printIndents();
-		writer.writeTagNameAndSeperator("lockedSpreadGate");
-		writer.write("\"0x");
-		intToHex(arpSettings.lastLockedSpreadGateParameterValue, buffer);
-		writer.write(buffer);
-		for (int i = 0; i < SPREAD_LOCK_MAX_SAVED_VALUES; i++) {
-			intToHex(arpSettings.lockedSpreadGateValues[i], buffer, 2);
-			writer.write(buffer);
-		}
-		writer.write("\"");
-		// Spread octave
-		writer.insertCommaIfNeeded();
-		writer.write("\n");
-		writer.printIndents();
-		writer.writeTagNameAndSeperator("lockedSpreadOctave");
-		writer.write("\"0x");
-		intToHex(arpSettings.lastLockedSpreadOctaveParameterValue, buffer);
-		writer.write(buffer);
-		for (int i = 0; i < SPREAD_LOCK_MAX_SAVED_VALUES; i++) {
-			intToHex(arpSettings.lockedSpreadOctaveValues[i], buffer, 2);
-			writer.write(buffer);
-		}
-		writer.write("\"");
 
 		if (output->type == OutputType::MIDI_OUT || output->type == OutputType::CV) {
-			writer.writeAttribute("gate", arpeggiatorGate);
-			writer.writeAttribute("rate", arpeggiatorRate);
+			writer.writeAttribute("gate", arpSettings.gate);
+			writer.writeAttribute("rate", arpSettings.rate);
 			// Community Firmware parameters (always write them after the official ones, just before closing the parent
 			// tag)
-			writer.writeAttribute("noteProbability", arpeggiatorNoteProbability);
-			writer.writeAttribute("ratchetProbability", arpeggiatorRatchetProbability);
-			writer.writeAttribute("ratchetAmount", arpeggiatorRatchetAmount);
-			writer.writeAttribute("sequenceLength", arpeggiatorSequenceLength);
-			writer.writeAttribute("rhythm", arpeggiatorRhythm);
-			writer.writeAttribute("spreadVelocity", arpeggiatorSpreadVelocity);
-			writer.writeAttribute("spreadGate", arpeggiatorSpreadGate);
-			writer.writeAttribute("spreadOctave", arpeggiatorSpreadOctave);
+			writer.writeAttribute("noteProbability", arpSettings.noteProbability);
+			writer.writeAttribute("bassProbability", arpSettings.bassProbability);
+			writer.writeAttribute("chordProbability", arpSettings.chordProbability);
+			writer.writeAttribute("ratchetProbability", arpSettings.ratchetProbability);
+			writer.writeAttribute("ratchetAmount", arpSettings.ratchetAmount);
+			writer.writeAttribute("sequenceLength", arpSettings.sequenceLength);
+			writer.writeAttribute("chordPolyphony", arpSettings.chordPolyphony);
+			writer.writeAttribute("rhythm", arpSettings.rhythm);
+			writer.writeAttribute("spreadVelocity", arpSettings.spreadVelocity);
+			writer.writeAttribute("spreadGate", arpSettings.spreadGate);
+			writer.writeAttribute("spreadOctave", arpSettings.spreadOctave);
```

Comment:
I think this should be done with a method on `ArpSettings`, so it can be shared. Something like `ArpSettings::writeMidiSettingsToFile(Serializer& writer)`.

Reply:
Similarly, the non-MIDI stuff would be like `ArpSettings::writeCommonSettingsToFile` or something like that.